### PR TITLE
added hw 5.1 solution

### PR DESCRIPTION
**Pull request by programmer28**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+public class Car implements Cloneable {
+    private String name;
+    private int speed;
+    private Engine engine;
+    private int age;
+    private String color;
 
+    @Override
+    protected Car clone() throws CloneNotSupportedException {
+        Car car = (Car)super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: can we solve without super.clone()?

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+public class Engine implements Cloneable {
+    private String name;
+    private double power;
+    private double price;
+    private int weight;
+    private boolean type;
 
+    @Override
+    protected Engine clone() throws CloneNotSupportedException {
+        return (Engine)super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java]: same

Comment (programmer28) [reply src/main/java/core/basesyntax/Engine.java]: same

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (programmer28) [reply src/main/java/core/basesyntax/Car.java]: Okay, I`ve changed it

**Pull request closed without merging**